internal/nodes/avi_model_evh_nodes.go

```diff
@@ -545,9 +546,12 @@ func (o *AviObjectGraph) ConstructAviL7SharedVsNodeForEvh(vsName, key string, ro
 	avi_vs_meta.ApplicationProfile = utils.DEFAULT_L7_APP_PROFILE
 	avi_vs_meta.NetworkProfile = utils.DEFAULT_TCP_NW_PROFILE
 	avi_vs_meta.EVHParent = true
-
-	vrfcontext := lib.GetVrf()
-	avi_vs_meta.VrfContext = vrfcontext
+	if lib.GetT1LRPath() != "" {
+		vrfcontext = ""
+	} else {
+		vrfcontext = lib.GetVrf()
+		avi_vs_meta.VrfContext = vrfcontext
```

Comment:
we seem to be changing the vrfcontext for the VS node based on the Tier1 LR, so do we need to have the vrfcontext accomodated in the VSNode's CalculateChecksum as well?

Reply:
I would think no. The reason is the following:
- the vrfcontext is always going to set to "" for the NSX-T cloud.
- the vrfcontext will behave the same way for all other cloud types.

If in the NSX-T cloud, the vrfcontext will never transition back to a non-empty value in layer 2.